packages/react-native/lib/credentials/credentialHooks.js

```diff
@@ -134,11 +134,17 @@ export async function getCredentialStatus(credential) {
   }
   return CREDENTIAL_STATUS.INVALID;
 }
+
+export let cachedCredentialStatus = {};
+
 export function useGetCredentialStatus({credential}) {
-  const [status, setStatus] = useState(CREDENTIAL_STATUS.PENDING);
+  const [status, setStatus] = useState(
+    cachedCredentialStatus[credential.id] || CREDENTIAL_STATUS.PENDING,
+  );
 
   useEffect(() => {
     getCredentialStatus(credential).then(response => {
+      cachedCredentialStatus[credential.id] = response;
```

Comment:
do we ever refresh this?

Reply:
Yes, if a cached value is found it will be displayed, but still fetch the current value from network and update it if its different